multinomial.hhs

```diff
@@ -0,0 +1,80 @@
+/**
+ * @author Jianan Lin (林家南)
+ * @param input - a series of numbers a1, ... an, or an array / matrix / tensor
+ * @returns - a number = (a1 + ... + an)! / (a1! * ... * an!)
+ * 
+ */
+
+
+function multinomial(input) {
+
+    *import math: is_number
+    *import math: flatten
+    *import math: deep_copy
+
+    let length = arguments.length;
+
+    // if there are no arguments, there is an error
+    if (length === 0) {
+        throw new Error('No argument given');
+    }
+
+    // if we input a series of numbers
+    if (is_number(input)) {
+        for (let i = 0; i < length; i++) {
+            if (arguments[i] < 0 || parseInt(arguments[i]) !== arguments[i]) {
+                throw new Error('Arguments must be non-negative integers');
+            }
+        }
+
+        let sum = 0;
+        for (let i = 0; i < length; i++) {
+            sum += arguments[i];
+        }
+
+        let result = private_factorial(sum);
+        for (let i = 0; i < length; i++) {
+            result /= private_factorial(arguments[i]);
+        }
+
+        return result;
+    }
+
+    else {
+        if (length !== 1) {
+            throw new Error('Wrong argument number');
+        }
+
```

Comment:
Right before declaring the variables for the matrix inputs, we should type check the arguments, similar to line 23-26. This is to ensure that if the user has a large amount of code written, and they make an error, it happens during runtime and they have an idea of where and why it occurs with the exception. Also, it'll be a new thing for the styling guide - but we should add what function name the exception comes from. 

```suggestion
if ( !(Array.isArray(arguments[0])) && !(arguments[0] instanceof Mat) && !(arguments[0] instanceof Tensor)) {
    throw new Error('Exception occurred in multinomial - argument is of length 1, not a number, but is not a Mat, Tensor or JS Array');
}
```
This is probably a little extra, but Lidangzzz mentioned type checking everything to be safe as we grow larger. Other than that it looks perfect!

Also I fixed factorial - my bad lol, you can use *import math:factorial if you want to, either way is ok